mongoengine/base/datastructures.py

```diff
@@ -212,7 +214,8 @@ class EmbeddedDocumentList(BaseList):
     def __match_all(cls, i, kwargs):
         items = kwargs.items()
         return all([
-            getattr(i, k) == v or unicode(getattr(i, k)) == v for k, v in items
+            getattr(i, k) == v or six.text_type(getattr(i, k)) == v
+            for k, v in items
         ])
```

Comment:
Thanks for paying attention to this. I rewrote this completely to make it easier to comprehend.